cpp/src/strings/strip.cu

```diff
@@ -83,15 +70,14 @@ std::unique_ptr<column> strip(
 
   auto const d_column = column_device_view::create(input.parent(), stream);
 
-  // this utility calls the strip_fn to build the offsets and chars columns
-  auto children = cudf::strings::detail::make_strings_children(
-    strip_fn{*d_column, side, d_to_strip}, input.size(), stream, mr);
+  auto result = rmm::device_uvector<string_index_pair>(input.size(), stream);
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::counting_iterator<size_type>(0),
+                    thrust::counting_iterator<size_type>(input.size()),
+                    result.begin(),
+                    strip_transform_fn{*d_column, side, d_to_strip});
 
-  return make_strings_column(input.size(),
-                             std::move(children.first),
-                             std::move(children.second),
-                             input.null_count(),
-                             cudf::detail::copy_bitmask(input.parent(), stream, mr));
+  return make_strings_column(result.begin(), result.end(), stream, mr);
```

Comment:
Does this change behavior? On first look, it seems to no longer preserve nulls, instead replacing them with empty strings. Is there a test covering this?

Reply:
Thanks! Just for my understanding: There is a difference between an empty `string_index_pair` pointing to `nullptr` and one pointing to actual data? Does this mean to create a column of empty strings, you need an allocation for the (albeit empty) strings?

Reply:
This applies only to using this factory function for the purposes of building the null-mask as explained here https://docs.rapids.ai/api/libcudf/stable/group__column__factories.html#ga17a8df722294a0856bd13641660976ef
You can specify  `string_index_pair{"",0}` for an empty string (like in this case) which does not require allocating any memory.